vtex-apps/return-app
# V3 committed items

This code finishes the process to calculate the processed items (items that cannot be included in a new return request either because they were included in another RMA or they have been included in a invoice type Input). The logic is the following (all is happening on `node/utils/createOrdersToReturnSummary.ts`:

1. The function searches for all the RMA that have been created for the order in context;
2. From the RMA, it catches the invoice number for that RMA - if it is already refunded;
3. From the RMA, it catches all the items included in that RMA;
4. From the invoices in the Order payload, it filters out all the invoices that were created by the RMA app (RMAs which the user selected to be refunded via credit card, or RMAs which the store doesn't allow the user to select the refund method AND the purchase was made using card - This process will be implemented in the end of the admin flow);
5. Using the item index, it calculates the quantity committed to be returned in other RMAs;
6. It adds up the quantity committed to be returned AND the quantity invoiced for every item. It it's greater than 0, it adds a new object inside the `processedItems` list.

### How to test it:
1. In this [workspace](https://rmav3createrma--powerplanet.myvtex.com), creates a new RMA;
7. After creating it, go back to the order list page and REFRESH the page. The item you just returned should be unavailable to be returned again.

__For now, we need to refresh the page to get the most update state. Our query to get the orders are cached and it keeps the previous value. We will have to deal with that later.__
